x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/EvaluatorImplementer.java

```diff
@@ -974,12 +977,114 @@ public String closeInvocation() {
         }
     }
 
+    private record BlockArrayProcessFunctionArg(TypeName componentType, String name) implements ProcessFunctionArg {
```

Comment:
I know this is private, but I think a line or two of Javadoc about what it's for would still be helpful.

Reply:
I'll add javadoc.